src/Sylius/Component/Product/Model/Product.php

```diff
@@ -440,22 +448,49 @@ protected function createTranslation(): ProductTranslationInterface
     }
 
     /**
-     * {@inheritdoc}
+     * @param ProductAttributeValueInterface $attributeValue
+     * @param string $localeCode
+     * @param string|null $fallbackLocaleCode
+     *
+     * @return AttributeValueInterface
      */
     private function getAttributeInDifferentLocale(
         ProductAttributeValueInterface $attributeValue,
-        ?string $localeCode
+        string $localeCode,
+        ?string $fallbackLocaleCode = null
     ): AttributeValueInterface {
-        if (!$this->hasAttributeByCodeAndLocale($attributeValue->getCode(), $localeCode)) {
+        if (!$this->hasNotEmptyAttributeByCodeAndLocale($attributeValue->getCode(), $localeCode)) {
+            if (
+                null !== $fallbackLocaleCode &&
+                $this->hasNotEmptyAttributeByCodeAndLocale($attributeValue->getCode(), $fallbackLocaleCode))
+            {
+                return $this->getAttributeByCodeAndLocale($attributeValue->getCode(), $fallbackLocaleCode);
+            }
+
             return $attributeValue;
         }
 
-        $attributeValueInDifferentLocale = $this->getAttributeByCodeAndLocale($attributeValue->getCode(), $localeCode);
-        if ('' === $attributeValueInDifferentLocale->getValue()
-            || null === $attributeValueInDifferentLocale->getValue()) {
-            return $attributeValue;
+        return $this->getAttributeByCodeAndLocale($attributeValue->getCode(), $localeCode);
+    }
+
+    /**
+     * @param string $attributeCode
+     * @param string $localeCode
+     *
+     * @return bool
+     */
+    private function hasNotEmptyAttributeByCodeAndLocale(string $attributeCode, string $localeCode): bool
+    {
+        $attributeValue = $this->getAttributeByCodeAndLocale($attributeCode, $localeCode);
+
+        if (null === $attributeValue) {
+            return false;
+        }
+
+        if (empty($attributeValue->getValue())) {
```

Comment:
I guess we do not want it to be an `empty` call.

> The following things are considered to be empty:
>  - "" (an empty string)
>  - 0 (0 as an integer)
>  - 0.0 (0 as a float)
>  - "0" (0 as a string)
>  - NULL
>  - FALSE
>  - array() (an empty array)
>  - $var; (a variable declared, but without a value)

Reply:
I'm afraid we're missing the tests for attributes that have values existing in base locale, but these values are considered empty (especially for booleans or integers).

Like with base locale `fr_FR` and current locale `pl_PL` and boolean attribute, what happens if:
 - boolean attribute has `false` value in `fr_FR` but `true` in `pl_PL`
 - integer attribute has `0` value in `fr_FR` but `1` in `pl_PL`